worlds/alttp/Options.py

```diff
@@ -110,13 +110,28 @@ class WorldState(Choice):
     option_inverted = 2
 
 
-class Bosses(Choice):
-    option_vanilla = 0
-    option_simple = 1
+class Bosses(TextChoice):
+    display_name = "Boss Shuffle"
+    option_none = 0
+    option_basic = 1
     option_full = 2
     option_chaos = 3
     option_singularity = 4
 
+    requires_plando = True
+
+    def verify(self, plando_options):
+        if isinstance(self.value, int):
+            return
+        from Generate import PlandoSettings
+        if not(PlandoSettings.bosses & plando_options):
+            options = self.value.split(";")
+            for option in options:
+                if option in self.options:
+                    self.value = Bosses.options[option]
+                    return
```

Comment:
i don't understand this part

Reply:
The format for plando bosses for LTTP is `Location-Boss;Location-Boss;remaining_shuffle` so this will check if the string contains a valid option name from the supplied choices for it to use before erroring out.

Reply:
maybe i'm stupid, but
* if you put "invalid;valid;invalid" there, it will ignore the two invalid and return the valid without telling the user, and
* i don't see a non-error return if PlandoSettings.bosses & plando_options is True

?